Kernel/Types/analyze.cpp

```diff
@@ -514,20 +514,23 @@ static string hundreds[10]= {"",  "c",  "cc",  "ccc",  "cd",
                              "d", "dc", "dcc", "dccc", "cm"};
 
 string
-roman_nr (int nr) {
-  if (nr < 0) return "-" * roman_nr (-nr);
+roman_nr (int32_t nr) {
   if (nr == 0) return "o";
-  if (nr > 1000) return "m" * roman_nr (nr - 1000);
   if (nr == 1000) return "m";
   if (nr == 999) return "im";
   if (nr == 499) return "id";
+  if (nr >= 10000) return "<unspecified>";
+  if (nr <= -10000) return "<unspecified>";
```

Comment:
this bound should be set to 3999, or output of 4001 is incorrect

Reply:
Fixed.